scipy/stats/_multivariate.py

```diff
@@ -4222,17 +4221,14 @@ def rvs(self, m, n, size=1, random_state=None):
         M, m, n, _, _, mncond = self._process_parameters(m, n)
 
         random_state = self._get_random_state(random_state)
-        try:
-            # Generator is only available in numpy >= 1.17 while
-            # `multivariate_hypergeometric` distribution is only
-            # available in numpy >= 1.18
-            if (isinstance(random_state, np.random.Generator) and
-                np.__version__ >= '1.18'):  # noqa: E129
-                out = random_state.multivariate_hypergeometric(m, n,
-                                                               size)
-                return _squeeze_output(out)
-        except AttributeError:
-            pass
+        # Generator is only available in numpy >= 1.17 while
+        # `multivariate_hypergeometric` distribution is only
+        # available in numpy >= 1.18
+        if (np.__version__ >= '1.18' and
```

Comment:
Here a minor comment, we can use pep-440 compliant version comparisons. Numpy both has this implemented now:
https://github.com/numpy/numpy/blob/b7c27bd2a3817f59c84b004b87bba5db57d9a9b0/numpy/lib/tests/test__version.py
so maybe something like: 
```
    # put import at top
    from numpy.lib import NumpyVersion
   ...
   # then here in line: 
    if (NumpyVersion(np.__version__) >= '1.18.0' and
   # the existing short circuiting should take care of the false case
```

Reply:
Something like this is needed, because the current version check is not correct.   `np.__version__` >= "1.18" will be true for older versions where the minor number is a single digit greater than 1,  such as "1.5.0".

Reply:
Fixed, @rlucas7. Thanks!

Reply:
IIUC the concensus in this PR is to remove the changes b/c of an upstream issue w/numpy's generator and braodcasting, so this is no longer an issue. Marking resolved. If I've misunderstood and this warrants reopening, please do so.